apps/prairielearn/src/middlewares/authzCourseOrInstance.sql

```diff
@@ -1,6 +1,20 @@
 -- BLOCK select_authz_data
+WITH
+  access_mode AS (
+    SELECT
+      mode,
+      mode_reason
+    FROM
+      ip_to_mode ($ip, $req_date, $user_id)
+  )
 SELECT
-  coalesce($req_mode, ip_to_mode ($ip, $req_date, $user_id)) AS mode,
+  coalesce($req_mode, (access_mode.mode)) AS mode,
+  (
+    CASE
+      WHEN $req_mode IS NOT NULL THEN 'Override'
```

Comment:
🤷 we could also leave this as `NULL`? We really only care whether or not the value is `PrairieTest`; what the "or not" value is doesn't really matter.

Reply:
As I commented elsewhere, I think it would be cleaner to have `mode_reason` be an enum with a carefully thought-through list of possible values.

Reply:
Switched to an enum in a8f961a.